### PR TITLE
dosbox-staging: 0.80.1 -> 0.81.1

### LINKED ISSUE #293911: Update request: dosbox-staging 0.80.1 → 0.81.0
- Package name: dosbox-staging
- Latest released version: 0.81.0

- Current version on the unstable channel: 0.80.1
- Current version on the stable/release channel: 0.80.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@JoshuaFern @AndersonTorres 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by JohnRTitor**
## Description of changes
- Closes https://github.com/NixOS/nixpkgs/issues/293911
- remove patches that are now upstream

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/do/dosbox-staging/package.nix:47]: Remove `fetchpatch` from the input set too.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 310664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dosbox-staging</li>
  </ul>
</details>

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:47]: Done. Thanks.

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (keenanweaver) [inline pkgs/by-name/do/dosbox-staging/package.nix:96]: Can we replace this with the [upstream desktop file](https://github.com/dosbox-staging/dosbox-staging/blob/main/contrib/linux/dosbox-staging.desktop)?

Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix]: Good idea. Done :)

Comment (keenanweaver) [reply pkgs/by-name/do/dosbox-staging/package.nix]: We probably actually want to name it dosbox-staging.desktop, do a substituteInPlace of the desktop file to set the exec to be `dosbox-staging` and set meta.mainProgram to `dosbox-staging` so there are no conflicts in case users want all the different DOSBoxes available in nixpkgs.

Also, looks like the build is failing now:

```
error: builder for '/nix/store/94nln40gaff7smqn12pri1zz033di0gi-dosbox-staging-0.81.0.drv' failed with exit code 1;
       last 25 log lines:
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_3848.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_20866.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28591.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28592.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28593.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28594.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28595.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28596.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28597.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28598.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28600.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28601.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28603.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28604.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/mapping-wikipedia.org/CP_28606.TXT to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/mapping-wikipedia.org
       > Installing /build/source/contrib/resources/translations/de.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/en.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/es.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/fr.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/it.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/nl.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/pl.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > Installing /build/source/contrib/resources/translations/ru.lng to /nix/store/21p472dka60056ichsh3xwdjrk5im8d7-dosbox-staging-0.81.0/share/dosbox-staging/translations
       > cp: cannot stat 'contrib/linux/dosbox-staging.desktop': No such file or directory
       > /nix/store/558iw5j1bk7z6wrg8cp96q2rx03jqj1v-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/94nln40gaff7smqn12pri1zz033di0gi-dosbox-staging-0.81.0.drv'.
error: 1 dependencies of derivation '/nix/store/f049fqaxblcqbax04rfqlpl0p4677pk9-review-shell.drv' failed to build
```

**Comment by keenanweaver** [comment]
0.81.1 is now out: https://github.com/dosbox-staging/dosbox-staging/releases/tag/v0.81.1

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix]: Fixed. Thanks.

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/do/dosbox-staging/package.nix:83]:
```suggestion
    install -Dm644 $src/contrib/linux/dosbox-staging.desktop $out/share/applications/
```

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/do/dosbox-staging/package.nix:29]:
```suggestion
  gitUpdater,
```

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/do/dosbox-staging/package.nix:100]:
```suggestion
  passthru.updateScript = gitUpdater {
    rev-prefix = "v";
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:29]: I tested `nix-update-script` and it seems to do its job fine, even the nixpkgs manual recommends nix update script. Is there a benefit of switching over to `gitUpdater`?

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:83]: Will do!

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:100]: @AndersonTorres https://github.com/NixOS/nixpkgs/pull/310664#discussion_r1597795168

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (AndersonTorres) [reply pkgs/by-name/do/dosbox-staging/package.nix:29]: 0. I would not say the manual recommends it, it just says it is common.
1. `gitUpdater` is in-house, while nix-update-script is (a wrapper for) an external tool, https://github.com/Mic92/nix-update
2. I am not sure if we should rely on a tool that explicitly say it is "better suited for interactive usage that might require user-intervention".

Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:29]: I don't think we shouldn't use a tool just because it's not "in-house". nix-update-script is smart enough to detect things like `rev-prefix`, on its own and more flexible. 

We already use tools like `nixpkgs-review` in nixpkgs to review PRs. It shouldn't matter whether it's external. The developer @Mic92 is well known contributor to Nixpkgs.

The disclaimer indeed has that warning, however nix-update-script is used by a lot of packages and so far I have not noticed any breakage. Every PR on Nixpkgs is reviewed by a maintainer anyway, and merged by a well trusted commiter (except for nixpkgs-merge-bot, we have that now). It's not like someone's opening a Pandora's box. 

I would like a third opinion on this, if possible.

**Comment by romatthe** [comment]
I believe 0.81.0 [includes support](https://dosbox-staging.github.io/releases/release-notes/0.81.0/#supercharged-image-and-video-capturing) for using `zlib-ng` during video capture for improved performance, so that should probably be included as a dependency now.

**Comment by JohnRTitor** [comment]
Done. Thanks.

**Comment by romatthe** [comment]
Another minor note: when I was building 0,81.0 myself a few days ago, I noticed that the version string in several locations was incorrect. See https://github.com/dosbox-staging/dosbox-staging/issues/3521#issuecomment-2103626719

Not sure if this is still an issue in 0.81.1 (at work atm) or if this is even worth trying to correct, but I figured I'd mention it.

**Comment by JohnRTitor** [comment]
image

Looks fine on my end.

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 310664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dosbox-staging</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]
I've no opinion on the nix-update-script stuff, but everything else seemingly is working great for me. Thanks for your effort!

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (AndersonTorres) [reply pkgs/by-name/do/dosbox-staging/package.nix:30]: Do you want to add yourself to maintainers?

Comment (AndersonTorres) [reply pkgs/by-name/do/dosbox-staging/package.nix:30]: Do you want to add yourself to `meta.maintainers`?

Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:30]: Nope.

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (keenanweaver) [inline pkgs/by-name/do/dosbox-staging/package.nix:113]:
```suggestion
    mainProgram = "dosbox-staging";
    maintainers = with lib.maintainers; [
```

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (keenanweaver) [inline pkgs/by-name/do/dosbox-staging/package.nix:98]:
```suggestion
    popd
    
    # Point desktop file to dosbox-staging
    substituteInPlace $out/share/applications/dosbox-staging.desktop \
    --replace-fail "Exec=dosbox" "Exec=dosbox-staging"
```

Comment (keenanweaver) [reply pkgs/by-name/do/dosbox-staging/package.nix:98]: Otherwise base DOSBox will run if the user has it in their path.

**Review by keenanweaver — CHANGES_REQUESTED** [review]

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:98]: https://github.com/NixOS/nixpkgs/blob/1a0c12d45e430c48b9c7038da373e265650ece6e/pkgs/by-name/do/dosbox-staging/package.nix#L93

the current postFixup creates a wrapper of `dosbox` which is under the hood `dosbox-staging`

Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:98]: The current process does not support both installations and will conflict. `dosbox` binary `dosbox` wrapper have the same name. It's hard to determine which one will launch first. We can, however, not rename the dosbox-staging at all, and create a wrapper instead at .dosbox-staging-wrapped. @AndersonTorres thoughts, I am not particularly experienced with the wrapping process.

Comment (AndersonTorres) [reply pkgs/by-name/do/dosbox-staging/package.nix:98]: I think we can discuss this independently, since it is more an enhancement than a bugfix.

@keenanweaver can you please open an issue ticket explaining your point, so we can track it?

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (AndersonTorres) [reply pkgs/by-name/do/dosbox-staging/package.nix:30]: > I don't think we shouldn't use a tool just because it's not "in-house".

I am not prohibiting anyone of using it. Indeed it can be used as a standalone tool.

I am just not wanting to use it without understanding its dark magic in an expression I will keep maintenance.

> nix-update-script is smart enough to detect things like rev-prefix, on its own and more flexible.

Well, not to being offensive, but if the upstream developer does not treat the tool as suitable for batch usage, I would not use it as a batch tool in production. 
Who will take responsibility in case of failure?

That being said, nix-update itself respects the update script already defined by the package. It looks  like an extra point to me.

> It shouldn't matter whether it's external.

The internal tool is already present in the Nixpkgs tree, not requiring an extra download trip and reducing the dependency counting.
Well, this is not a strong argument since no update script can run offline - but can be useful if running a batch updater.

> Every PR on Nixpkgs is reviewed by a maintainer anyway, and merged by a well trusted commiter

Famous last words...

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (keenanweaver) [reply pkgs/by-name/do/dosbox-staging/package.nix:98]: Just opened. https://github.com/NixOS/nixpkgs/issues/312101

Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:98]: Resolving this, then.

**Review by keenanweaver — APPROVED** [review]

File: pkgs/by-name/do/dosbox-staging/package.nix
Comment (JohnRTitor) [reply pkgs/by-name/do/dosbox-staging/package.nix:30]: I was hoping if we could reach an agreement, but well, it is what it is.
If upstream versioning scheme changes in the foreseeable future, nix-update-script would still work, because that's how flexible it is.

I am not planning to mass update packages using the tool, nor I have seen nixpkgs mass updates packages (unless in very specific cases such as GNOME 46/issues).

Even if ryantm-r ends up auto bumping this, along with other packages in batch, I am positive that the PR will atleast be reviewed by a human.

Comment (AndersonTorres) [reply pkgs/by-name/do/dosbox-staging/package.nix:30]: The agreement is already done. I will defer to your judgement and depart from all responsibility.

There are way more problems to be solved before losing energy and sleep time on what is the best dark magic updater.

**Pull request merged**
